### ISSUE TITLE
use test human.fa and bonbo.fa cannot use JustOrthologs

### ISSUE THREAD

**Issue by wangjie1201** [ridgelab/JustOrthologs#4]

**Comment by wangjie1201** [comment]
If i put the command that you offered,there is a error,File "justOrthologs.py", line 7.
![image](https://user-images.githubusercontent.com/48744177/59978242-fd99cd80-960c-11e9-8544-c8d6505f31a0.png)

I wish your help
thank you!

**Comment by ridgelab** [comment]
Hello,
The error seems to indicate that justOrthologs.py is an html file for you. Did you you clone JustOrthologs into your local directory using git clone? If that doesn't work, do you mind sending me the command that you used and the justOrthologs.py script so I can ensure that you have the correct version?

**Comment by wangjie1201** [comment]
Hello,
Thank you for give me the solution.The error had solved.But i have another question， if I want to use your gff3_parser.py to make each CDS region followed by an asterisk(*),i need to give the FASTA header lines must be formatted in one of two ways:>NC_###### or >gi|####|ref|NC_#####?
If i have the fasta header lines,such as Potri####,what should i do?
Thanks again for your reply





------------------ 原始邮件 ------------------
发件人: "ridgelab"<notifications@github.com>;
发送时间: 2019年6月25日(星期二) 凌晨1:59
收件人: "ridgelab/JustOrthologs"<JustOrthologs@noreply.github.com>;
抄送: "675103467"<675103467@qq.com>;"State change"<state_change@noreply.github.com>;
主题: Re: [ridgelab/JustOrthologs] use test human.fa and bonbo.fa cannotuse JustOrthologs (#4)




Hello,
 The error seems to indicate that justOrthologs.py is an html file for you. Did you you clone JustOrthologs into your local directory using git clone? If that doesn't work, do you mind sending me the command that you used and the justOrthologs.py script so I can ensure that you have the correct version?

**Comment by ridgelab** [comment]
I'm glad that worked for you.
Currently we don't support Potri numbers, but you should be able to use >Potri#### as the header line as long as the Potri# is used the same way as the NC_# would be used in the gff3 file. The purpose of the gff3 parser is to extract the CDS regions from the fasta file and combine CDS regions belonging to the same gene by putting an * between each CDS region. For example, if gene 1 had one CDS region and gene 2 had three CDS regions, the output file would look something like this:
>gene 1
CDS1*
>gene 2
CDS1*CDS2*CDS3*

If you are able to get an output file in the same format (with or without the gff3_parser.py) then you can sort the file by the number of CDS regions (using sortFastaBySeqLen.sh) and run JustOrthologs.py on it. 
Hope that helps.

**Comment by ridgelab** [comment]
Looks like the * italicized the text. Let me try again for the output file:

`>gene 1
CDS1[asterisk]
>gene 2
CDS1[asterisk]CDS2[asterisk]CDS3[asterisk]`

**Comment by wangjie1201** [comment]
Thank you
I will try it.You're so good 




------------------ 原始邮件 ------------------
发件人: "ridgelab"<notifications@github.com>;
发送时间: 2019年6月27日(星期四) 凌晨3:59
收件人: "ridgelab/JustOrthologs"<JustOrthologs@noreply.github.com>;
抄送: "675103467"<675103467@qq.com>;"State change"<state_change@noreply.github.com>;
主题: Re: [ridgelab/JustOrthologs] use test human.fa and bonbo.fa cannotuse JustOrthologs (#4)




Looks like the * italicized the text. Let me try again for the output file:
 
`>gene 1
 CDS1[asterisk]
  
gene 2
 CDS1[asterisk]CDS2[asterisk]CDS3[asterisk]`

**Comment by wangjie1201** [comment]
I'm sorry to bother your again. Others can do it ,but i have another one can not do it.The file is attached.
link: https://pan.baidu.com/s/1MLQK1bBzYZZPbC-DYi9Bkg   Extract the code:3fbg 
I don't know what went wrong.





------------------ 原始邮件 ------------------
发件人: "ridgelab"<notifications@github.com>;
发送时间: 2019年6月27日(星期四) 凌晨3:59
收件人: "ridgelab/JustOrthologs"<JustOrthologs@noreply.github.com>;
抄送: "675103467"<675103467@qq.com>;"State change"<state_change@noreply.github.com>;
主题: Re: [ridgelab/JustOrthologs] use test human.fa and bonbo.fa cannotuse JustOrthologs (#4)




Looks like the * italicized the text. Let me try again for the output file:
 
`>gene 1
 CDS1[asterisk]
  
gene 2
 CDS1[asterisk]CDS2[asterisk]CDS3[asterisk]`

**Comment by wangjie1201** [comment]
I tried again, and it worked. Thank you very much for your help




------------------ 原始邮件 ------------------
发件人: "ridgelab"<notifications@github.com>;
发送时间: 2019年6月27日(星期四) 凌晨3:59
收件人: "ridgelab/JustOrthologs"<JustOrthologs@noreply.github.com>;
抄送: "675103467"<675103467@qq.com>;"State change"<state_change@noreply.github.com>;
主题: Re: [ridgelab/JustOrthologs] use test human.fa and bonbo.fa cannotuse JustOrthologs (#4)




Looks like the * italicized the text. Let me try again for the output file:
 
`>gene 1
 CDS1[asterisk]
  
gene 2
 CDS1[asterisk]CDS2[asterisk]CDS3[asterisk]`

**Comment by ridgelab** [comment]
I'm glad that you were able to get it to work. Good luck with your analyses.

**Comment by wangjie1201** [comment]
I use gff3_parser.py ,but the output.fasta like this:
gene 1
CDS1*
gene 2 
CDS1*
gene 2 
CDS2*
gene 2 
CDS3*
I donnot know why
